src/cpp/src/tokenizer.cpp

```diff
@@ -137,7 +169,7 @@ class Tokenizer::TokenizerImpl {
 
     // To change the adding special tokens mode we use a statefull subgraph,
     // this flag holds the current state value of the CompiledModel.
-    ov::AnyMap m_state_flags;
+    std::unordered_map<ov::InferRequest*, ov::AnyMap> m_request_to_state_flags;
```

Comment:
Can you try std::reference_wrapper instead of ptr? They can't be null

Reply:
`std::reference_wrapper` is not hashable, which adds complexity. If you still want to go this route, we can merge it in the next PR.